xendit_pos/models/xendit_client.py

```diff
@@ -28,88 +31,26 @@ def _get_xendit_payment_method(self):
             )
 
-    def _get_xendit_api_key(self):
+    def _get_xendit_secret_key(self):
         xendit_payment_method = self._get_xendit_payment_method(self)
         return xendit_payment_method.xendit_pos_secret_key
 
-    def _generate_address(data):
-        addresses = []
-
-        if data["client"] == None:
-            return addresses
-
-        address = {
-            "city": data["client"]["city"],
-            "country": data["client"]["country_id"][1],
-            "postal_code": data["client"]["zip"],
-            "state": data["client"]["state_id"][1],
-            "street_line1": data["client"]["street"]
-        }
-        addresses.append(address)
-        return addresses
-
-    def _generate_customer(self, data):
-
-        if data["client"] == None:
-            return {}
-
-        return {
-            "given_names": data["client"]["name"],
-            "email": data["client"]["email"],
-            "mobile_number": data["client"]["phone"],
-            "address": self._generate_address(data)
-        }
-
-    def _generate_notification(self):
-        notificationReferences = {
-            "whatsapp",
-            "sms",
-            "email",
-            "viber"
-        }
-        return {
-            "invoice_created": notificationReferences,
-            "invoice_reminder": notificationReferences,
-            "invoice_paid": notificationReferences,
-            "invoice_expired":  notificationReferences
-        }
-
-    def _generate_items(data):
-        items = []
-
-        for orderline in data["orderlines"]:
-            item = {
-                "name": orderline["product_name"],
-                "price": orderline["price"],
-                "quantity": orderline["quantity"]
-            }
-            items.append(item)
-        return items
-
-    def _encodeAPIKey(self):
-        secret_key = self._get_xendit_api_key(self) + ":"
-        secret_key_bytes = secret_key.encode('ascii')
-        base64_bytes = base64.b64encode(secret_key_bytes)
-        return base64_bytes.decode('ascii')
-
     def _generate_header(self):
-        return {
-            'content-type': 'application/json',
-            'x-plugin-name': 'ODOO_POS',
-            'x-plugin-version': '1.0',
-            'Authorization': 'Basic ' + self._encodeAPIKey(self)
-        }
+        return self.dataUtils.generateHeader(
+            self._get_xendit_secret_key(self)
+        )
 
     def _generate_payload(self, data):
-        customer = self._generate_customer(self, data)
+
+        customerObject = self.dataUtils.generateInvoiceCustomer(data['client'])
         return {
             "external_id": data["name"].split(" ")[1],
             "amount": data["total_rounded"],
             "currency": data["currency"]["name"],
-            "payer_email": customer["email"] if len(customer) > 0 else "test@example.com",
+            "payer_email": customerObject["email"] if len(customerObject) > 0 else "test@example.com",
```

Comment:
Don't put default value.
payer_email is optional.

Reply:
@candrasaputra Updated